frontend/talentsearch/src/js/components/workPreferencesForm/WorkPreferencesForm.tsx

```diff
@@ -201,6 +201,30 @@ export const WorkPreferencesApi: React.FunctionComponent = () => {
       id,
       user: data,
     }).then((result) => {
+      if (result.data?.updateUserAsUser) {
+        if (result.data?.updateUserAsUser?.isProfileComplete) {
+          sessionStorage.setItem("currentProfileStatus", "Complete");
+        } else {
+          sessionStorage.setItem("currentProfileStatus", "InComplete");
+        }
+        const preProfileStatus = sessionStorage.getItem("preProfileStatus");
+        const currentProfileStatus = sessionStorage.getItem(
+          "currentProfileStatus",
+        );
+
+        if (preProfileStatus === "InComplete") {
+          if (currentProfileStatus === "Complete") {
```

Comment:
After comparing the prev and current values, you should update the prev value to be the new value.